### PR TITLE
feature/remove-pfconfigpool2

### LINKED ISSUE #7945: Remove use of golang pfconfigpool
**Is your feature request related to a problem? Please describe.**
Remove use of golang pfconfigpool

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Remove the use of pfconfigpool

# Issue
fixes #7945

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add OpenAPI specification
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## Enhancements
* Remove the use of pfconfigpool

### CODE REVIEW

File: go/firewallsso/factory.go [hunk 1]
```diff
@@ -3,33 +3,43 @@ package firewallsso
...
-	f.typeRegistry["JSONRPC"] = reflect.TypeOf(&JSONRPC{}).Elem()
-	f.typeRegistry["JuniperSRX"] = reflect.TypeOf(&JuniperSRX{}).Elem()
-	f.typeRegistry["FamilyZone"] = reflect.TypeOf(&FamilyZone{}).Elem()
-	f.typeRegistry["CiscoIsePic"] = reflect.TypeOf(&CiscoIsePic{}).Elem()
-	f.typeRegistry["ContentKeeper"] = reflect.TypeOf(&ContentKeeper{}).Elem()
+	f.typeRegistry = map[string]func() FirewallSSOInt{
+		"BarracudaNG":      newFirewallSSO[BarracudaNG],
+		"Iboss":            newFirewallSSO[Iboss],
+		"PaloAlto":         newFirewallSSO[PaloAlto],
+		"LightSpeedRocket": newFirewallSSO[FortiGate],
```
Comment (satkunas) [inline go/firewallsso/factory.go:32]: [LightSpeedRocket] ?

File: go/firewallsso/factory.go [hunk 2]
```diff
@@ -3,33 +3,43 @@ package firewallsso
...
+		"SmoothWall":       newFirewallSSO[FortiGate],
```
Comment (satkunas) [inline go/firewallsso/factory.go:33]: [SmoothWall] ?

File: go/plugin/coredns/pfdns/pfdns.go [hunk 3]
```diff
@@ -118,34 +112,13 @@ func (pf *pfdns) HasSecurityEvents(ctx context.Context, mac string) bool {
...
-						log.LoggerWContext(ctx).Error(err.Error())
-					}
-					time.Sleep(1 * time.Second)
-				}
-			}(ctx)
-		})
-	} else {
-		panic("Unable to obtain pfconfigpool lock in pfdns middleware")
-	}
+	// pf.pfdnsRefreshableConfig = pfconfigdriver.GetRefresh(ctx, "pfdnsRefreshableConfig").(*pfdnsRefreshableConfig)
```
Comment (satkunas) [inline go/plugin/coredns/pfdns/pfdns.go]: refresh not needed?

**Review by satkunas — CHANGES_REQUESTED** [review]

File: go/firewallsso/factory.go (hunk 1 above)
Comment (fdurand) [reply go/firewallsso/factory.go:32]: It's aliases

File: go/firewallsso/factory.go (hunk 2 above)
Comment (fdurand) [reply go/firewallsso/factory.go:33]: It's aliases

Comment (jrouzierinverse) [reply go/firewallsso/factory.go:33]: It was FortiGate previously

File: go/firewallsso/factory.go (hunk 1 above)
Comment (jrouzierinverse) [reply go/firewallsso/factory.go:32]: It was FortiGate previously

File: go/plugin/coredns/pfdns/pfdns.go (hunk 3 above)
Comment (jrouzierinverse) [reply go/plugin/coredns/pfdns/pfdns.go]: The refresh is done elsewhere in the code.

**Comment by fdurand** [comment]
Good to merge for me

**Pull request merged**